src/WalletConnectSigner.ts

```diff
@@ -0,0 +1,142 @@
+import type WalletConnect from "@walletconnect/client";
+import * as ethers from "ethers";
+import * as ethersBytes from "@ethersproject/bytes";
+import type {
+  Provider,
+  TransactionRequest,
+  TransactionResponse,
+} from "@ethersproject/abstract-provider";
+
+import {
+  Deferrable,
+  defineReadOnly,
+  resolveProperties,
+} from "@ethersproject/properties";
+import { _TypedDataEncoder } from "ethers/lib/utils";
+import type { TypedDataDomain, TypedDataField } from "@ethersproject/abstract-signer";
+
+export class WalletConnectSigner extends ethers.Signer {
+  public walletConnect: WalletConnect;
+
+  private _provider: ethers.providers.JsonRpcProvider;
+  constructor(
+    walletConnect: WalletConnect,
+    provider: ethers.providers.JsonRpcProvider,
+  ) {
+    super();
+    defineReadOnly(this, "provider", provider);
```

Comment:
You can do `readonly provider` in typescript. We can avoid having the variable set twice.

Reply:
yeah true!

Reply:
Unresolved :warning:

Reply:
This has been resolved